### PR TITLE
Transcripts: Refactor scroll to range textKit code

### PR DESCRIPTION
**Pull request by SergioEstevao**
| 📘 Part of: #1848  |  
|:---:|

Fixes # 

This PR just simplifies up the TextKit code used to scroll to specific ranges.
In the process also removes a console warning when opening the Transcript view for the first time
```
UITextView 0x10f06b600 is switching to TextKit 1 compatibility mode because its layoutManager was accessed. Break on void _UITextViewEnablingCompatibilityMode(UITextView *__strong, BOOL) to debug.
```

## To test

1. Start the app
2. Ensure you the transcripts FF enabled
3. Open an episode with transcripts. Ex: any episode of Cautionary Tales
4. Tap on play
5. Open the full screen player
6. Tap on Transcript button
7. Scroll around in the transcript
8. Tap on Search
9. Search for a word in the transcript
10. Go up and down on search results, make sure you cycle from the first to last result and vice versa
11. Ensure that scroll works correctly
12. Show and hide the keyboard by tapping on the Transcript text and ensure the position in scroll is kept

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Modules/Utils/Sources/PocketCastsUtils/Extensions/UI/UITextView+scrollToRange.swift [hunk 1]
```diff
@@ -4,34 +4,31 @@ import UIKit
...
-            ).inset(by: contentInset)
-            visibleRect = .init(x: visibleRect.origin.x, y: visibleRect.origin.y + (contentInset.bottom / 2), width: visibleRect.width, height: visibleRect.height)
+        // Calculate the rectangle to scroll to such that the finalRect is centered
+        var visibleRect = CGRect(
+            x: finalRect.origin.x - (bounds.width / 2) + (finalRect.width / 2),
+            y: finalRect.origin.y - (bounds.height / 2) + (finalRect.height / 2),
+            width: bounds.width,
+            height: bounds.height
+        ).inset(by: contentInset)
+        visibleRect = .init(x: visibleRect.origin.x, y: visibleRect.origin.y + (contentInset.bottom / 2), width: visibleRect.width, height: visibleRect.height)
```
Comment (bjtitus) [inline Modules/Utils/Sources/PocketCastsUtils/Extensions/UI/UITextView+scrollToRange.swift]: Could this be simplified to `visibleRect.origin.y += contentInset.bottom / 2`?

**Review by bjtitus — APPROVED** [review]
✅ Tested scrolling and searching and never saw an issue with either.

Left one small nit about the `visibleRect` y value modification.

File: Modules/Utils/Sources/PocketCastsUtils/Extensions/UI/UITextView+scrollToRange.swift (hunk 1 above)
Comment (SergioEstevao) [reply Modules/Utils/Sources/PocketCastsUtils/Extensions/UI/UITextView+scrollToRange.swift]: Done here: https://github.com/Automattic/pocket-casts-ios/pull/2155/commits/93985d1d4747a445ad557142c846a7bd5538de2e

**Pull request merged**